### PR TITLE
perlPackages.LatexIndent: init at 3.21

### PR DESCRIPTION
**Pull request by Tom-Hubrecht**
###### Description of changes

Initialize a package for latexindent.pl

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by RaitoBezarius — APPROVED** [review]
LGTM

**Comment by Tom-Hubrecht** [comment]
cc @RaitoBezarius

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -12904,6 +12904,32 @@ with self; {
...
+    version = "3.21";
+
+    src = fetchFromGitHub {
+      owner = "cmhughes";
+      repo = pname;
+      rev = "V${version}";
+      hash = "sha256-STXHOzsshyN7rc2VtJxxt6La4UPGpRYlMO8TX1Jd7pM=";
+    };
+
+    outputs = [ "out" ];
```
Comment (RaitoBezarius) [inline pkgs/top-level/perl-packages.nix:12918]:
```suggestion
```

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -12904,6 +12904,32 @@ with self; {
...
+    propagatedBuildInputs = [ FileHomeDir YAMLTiny ];
+
+    preBuild = ''
+      patchShebangs ./latexindent.pl
+    '';
+
+    meta = {
+      description = "Perl script to add indentation to LaTeX files";
+      homepage = "https://github.com/cmhughes/latexindent.pl";
+      license = with lib.licenses; [ gpl3Plus ];
```
Comment (RaitoBezarius) [inline pkgs/top-level/perl-packages.nix:12929]:
```suggestion
      license = with lib.licenses; gpl3Plus;
```

**Review by RaitoBezarius — APPROVED** [review]

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 229708` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>perl534Packages.LatexIndent</li>
    <li>perl534Packages.LatexIndent.devdoc</li>
    <li>perl536Packages.LatexIndent</li>
    <li>perl536Packages.LatexIndent.devdoc</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 229708` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>perl534Packages.LatexIndent</li>
    <li>perl536Packages.LatexIndent</li>
  </ul>
</details>

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**